File: Solutions/1402. Reducing Dishes/1402. Reducing Dishes.js
/**
 * @param {number[]} satisfaction
 * @return {number}
 */
var maxSatisfaction = function(s) {
    s.sort((a,b)=>a-b);
    var res=0,total=0;
    for(let i=s.length-1;i>=0&&s[i]+total>0;--i){
        total+=s[i];
        res+=total;
    }
    return res;
};

File: Solutions/1402. Reducing Dishes/1402. Reducing Dishes.go
func maxSatisfaction(s []int) int {
    sort.Ints(s)
    res,total :=0,0
    for i:=len(s)-1;i>=0&&s[i]+total>0;i--{
        total+=s[i]
        res+=total
    }
    return res
}